src/DynamoPackages/PackageLoader.cs

```diff
@@ -200,9 +202,45 @@ private void TryLoadPackageIntoLibrary(Package package)
             {
                 var dynamoVersion = VersionUtilities.PartialParse(DynamoModel.Version);
 
+                package.EnumerateAdditionalFiles();
+
+                // If the additional files contained an extension manifest, then request it be loaded.
+                var extensions = package.AdditionalFiles.Where(
+                    file => file.Model.Name.Contains("ExtensionDefinition.xml") || file.Model.Name.Contains("ViewExtensionDefinition.xml")).ToList();
+
+                AssemblyLoadContext pkgLoadContext = AssemblyLoadContext.Default;
+
+                bool isolateAnyPackage = DynamoModel.FeatureFlags?.CheckFeatureFlag("IsolateAnyPackage", false) ?? false;
+                bool isolateThisPackage = DynamoModel.FeatureFlags?.CheckFeatureFlag("IsolatePackage_" + package.Name, false) ?? false;
+                if (isolateAnyPackage || isolateThisPackage)
+                {
+                    if (extensions.Count() == 1)
+                    {
+                        var ext = extensions[0];
+                        string entryPoint = string.Empty;
+                        try
+                        {
+                            entryPoint = ExtensionLoader.GetExtensionPath(ext.Model.FullName);
+                            if (!string.IsNullOrEmpty(entryPoint))
+                            {
+                                pkgLoadContext = new PkgAssemblyLoadContext(package.Name + "@" + package.VersionName, Path.Combine(package.RootDirectory, entryPoint));
+                            }
+                        }
+                        catch (Exception ex)
+                        {
+                            Log(ex);
+                        }
+                    }
+                    else if (package.Header.node_libraries.Any())
+                    {
+                        string entryPoint = new AssemblyName(package.Header.node_libraries.First()).Name + ".dll";
```

Comment:
there could be more than one node_library right?

Reply:
I'll try the root package location first...then I'll circle back to this if needed